tests/test_snapshot_execution_time.py

```diff
@@ -0,0 +1,134 @@
+import time
+import os
+import pytest
+from bananalyzer.data.examples import get_training_examples
+from playwright.async_api import async_playwright
+
+IS_GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS") == "true"
+example_data = [
+    {
+        "id": "h4q2uwr0z0sVFM0q5AV7n",
+        "expected_page_to_image_time": 1.0,
+        "expected_page_to_text_time": 3.0,
+    },
+    {
+        "id": "7dhL1dij4AsT9qWCbWBHq",
+        "expected_page_to_image_time": 6.0,
+        "expected_page_to_text_time": 10.0,
+    },
+    {
+        "id": "CsjbrXOwtX1rRqggZALRB",
+        "expected_page_to_image_time": 1.0,
+        "expected_page_to_text_time": 2.0,
+    },
+    {
+        "id": "a0pJxHhxIHFKcoFjkORnG",
+        "expected_page_to_image_time": 1.0,
+        "expected_page_to_text_time": 1.5,
+    },
+    {
+        "id": "4Je6qSd4YFoyLxVZLQRb7",
+        "expected_page_to_image_time": 0.5,
+        "expected_page_to_text_time": 1.5,
+    },
+    {
+        "id": "u3fjwZRjKUEcvr8kkmy5v",
+        "expected_page_to_image_time": 2.0,
+        "expected_page_to_text_time": 5.0,
+    },
+    {
+        "id": "1JWoJWs3uZMt8Wa5ql6pr",
+        "expected_page_to_image_time": 3.5,
+        "expected_page_to_text_time": 4.2,
+    },
+    {
+        "id": "nxkcxrThdmaRX01YRXtho",
+        "expected_page_to_image_time": 2.0,
+        "expected_page_to_text_time": 5,
+    },
+    {
+        "id": "BBofvHlOP5C5aSVJknz1C",
+        "expected_page_to_image_time": 0.9,
+        "expected_page_to_text_time": 3.5,
+    },
+    {
+        "id": "7xlvZTTi21A1s7k3AoBOS",
+        "expected_page_to_image_time": 2.0,
+        "expected_page_to_text_time": 5.5,
+    },
+    {
+        "id": "ct6PuXzujbOlM9zaARUpa",
+        "expected_page_to_image_time": 3.0,
+        "expected_page_to_text_time": 12.0,
+    },
+    {
+        "id": "bwwko5J7aFk5K8qz61jBI",
+        "expected_page_to_image_time": 1.0,
+        "expected_page_to_text_time": 4,
+    },
+    {
+        "id": "24SLE3KnDhtOYYgIM4ote",
+        "expected_page_to_image_time": 1.0,
+        "expected_page_to_text_time": 3.5,
+    },
+    {
+        "id": "qtRibcsG6iq09TyGQoYhv",
+        "expected_page_to_image_time": 3.0,
+        "expected_page_to_text_time": 6.5,
+    },
+    {
+        "id": "RVotqLcMUyKXULUTqYCvm",
+        "expected_page_to_image_time": 2.0,
+        "expected_page_to_text_time": 2.2,
+    },
+]
+
+if not IS_GITHUB_ACTIONS:
+    all_examples = get_training_examples()
+    examples = [
+        {
+            "example": example,
+            "expected_page_to_image_time": data["expected_page_to_image_time"],
+            "expected_page_to_text_time": data["expected_page_to_text_time"],
+        }
+        for data in example_data
+        for example in all_examples
+        if example.id == data["id"]
+    ]
+else:
+    examples = []
```

Comment:
Very minor but negative conditions are a bit harder to read
```suggestion
if IS_GITHUB_ACTIONS:
    examples = []
else:
    all_examples = get_training_examples()
    examples = [
        {
            "example": example,
            "expected_page_to_image_time": data["expected_page_to_image_time"],
            "expected_page_to_text_time": data["expected_page_to_text_time"],
        }
        for data in example_data
        for example in all_examples
        if example.id == data["id"]
    ]

```